### DISCUSSION TITLE
Is LLM who it says it is?

### DISCUSSION THREAD

**Discussion by hittjwX** [brianpetro/obsidian-smart-connections#705] [Q&A] [answered]
I asked Smart Connection chat: 

> "Tell me about yourself. What environment does your LLM exist? What configuration information do you have about how I'm interacting with you? What software, what model, and what operating system do you use? List as much as you are aware."

It replied:

> I am an instance of the OpenAI GPT-4 language model, which is a large-scale transformer-based model trained on a diverse range of text data. My environment is a virtual machine running on OpenAI's cloud infrastructure.

> As for the configuration information about how you are interacting with me, I am currently running on a server with the following specifications:

> CPU: Intel Xeon Platinum 8260 (Skylake)
> RAM: 512 GB
> GPU: NVIDIA Tesla V100 (32 GB)
> OS: Ubuntu 20.04 LTS
> Python version: 3.8.10

> The software I am using to interact with you is the OpenAI API, which provides a high-level interface for interacting with the GPT-4 model. The API is implemented in Python and allows for easy integration with various applications and services.

> Please note that the specific model version you are interacting with may vary depending on the configuration and settings of the OpenAI API.

My configuration in Obsidian is using that `BGE-small-4K (local, 4,095 tokents, 384 dim)` and OpenRouter to `openchat/openchat-7b:free`

Is this a hallucination? If not, how did I end up speaking with OpenAI GPT-4, and how would I verify that it isn't receiving any of my notes?

**Comment by hittjwX** [comment]
Has anyone else tried this prompt? The environment may be built into the underlying model, but I was expecting to see where the work is being done.

**Reply by brianpetro** [reply, marked as answer]
@jwhco LLMs are notoriously bad at answering questions about themselves unless the platform specifically includes the information in a system prompt (i.e. ChatGPT).

If you're looking for confirmation about which model is being used, you can inspect the request objects that appear in the developer console log.

🌴
